tests/integration/simple_integration_test.go

```diff
@@ -0,0 +1,164 @@
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"io/ioutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var (
+	namespace = "k8ssandra"
+	releaseName string
+	releaseNameCluster string
+)
+
+var _ = Describe("Install the cluster", func() {
+	// TODO This assumes the kind was started with correctly setup cluster. Could we automate it here to provider proper e2e?
+
+	BeforeSuite(func() {
+		kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+		options := &helm.Options{KubectlOptions: kubectlOptions}
+
+		// Add traefik repo and update repos
+		helm.RunHelmCommandAndGetOutputE(GinkgoT(), options, "repo", "add", "traefik", "https://helm.traefik.io/traefik")
+		helm.RunHelmCommandAndGetOutputE(GinkgoT(), options, "repo", "update")
+
+		// Deploy traefik
+		// helm install traefik traefik/traefik -n traefik --create-namespace -f docs/content/en/docs/topics/ingress/traefik/kind-deployment/traefik.values.yaml
+		valuesPath, _ := filepath.Abs("../../docs/content/en/docs/topics/ingress/traefik/kind-deployment/traefik.values.yaml")
+		helm.RunHelmCommandAndGetOutputE(GinkgoT(), options, "install", "traefik", "traefik/traefik", "-n", "traefik", "--create-namespace", "-f", valuesPath)
+	})
+
+	BeforeEach(func() {
+		kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+		k8s.CreateNamespace(GinkgoT(), kubectlOptions, "k8ssandra")
+	})
+
+	AfterEach(func() {
+		kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+		options := &helm.Options{KubectlOptions: kubectlOptions}
+
+		// Verify everything was actually deleted.. I don't think Service is
+		helm.Delete(GinkgoT(), options, releaseNameCluster, true)
+		helm.Delete(GinkgoT(), options, releaseName, true)
+
+		// Delete all the resources from namespace
+		kubectlOptions = k8s.NewKubectlOptions("", "", "default")
+		k8s.DeleteNamespace(GinkgoT(), kubectlOptions, namespace)
+	})
+
+	Context("deploying", func() {
+		It("with default options", func() {
+			crdChartPath, err := filepath.Abs("../../charts/k8ssandra")
+			Expect(err).To(BeNil())
+
+			//kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+			//k8s.CreateNamespace(GinkgoT(), kubectlOptions, namespace)
+			kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+			options := &helm.Options{KubectlOptions: kubectlOptions}
+
+			releaseName = fmt.Sprintf(
+				"demo-%s", strings.ToLower(random.UniqueId()))
+			helm.Install(GinkgoT(), options, crdChartPath, releaseName)
+
+			// k8s module has no select by label
+			clientset, err := k8s.GetKubernetesClientFromOptionsE(GinkgoT(), kubectlOptions)
+			pods, _ := clientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "name=cass-operator"})
```

Comment:
@jeffbanks This is one of the areas that precipitated some of the recent discussions. Here we are using a generated, static client. In medusa-operator and in reaper-operator we use the controller-runtime dynamic client. 

On the one hand it is convenient that the `k8s` package from terratest already initializes the client for us. On the other hand, I prefer to stick with the controller-runtime client throughout. We use it extensively in both production and test code for operators.